plugins/modules/keycloak_authentication_required_actions.py

```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017, Eike Frost <ei@kefro.st>
+# Copyright (c) 2021, Christophe Gilles <christophe.gilles54@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_authentication_required_actions
+
+short_description: Allows administration of Keycloak authentication required actions.
+
+description:
+    - This module can register, update and delete required actions.
+    - It also filters out any duplicate required actions by their alias. The first ocurrence is preserved.
+
+version_added: 7.1.0
+
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: full
+
+options:
+    realm:
+        description:
+            - The name of the realm in which are the authentication required actions.
+        required: true
+        type: str
+    required_actions:
+        elements: dict
+        description:
+            - Authentication required action.
+        suboptions:
+            alias:
+                description:
+                    - Unique name of the required action.
+                required: true
+                type: str
+            config:
+                description:
+                    - Configuration for the required action.
+                type: dict
+            defaultAction:
+                default: false
```

Comment:
It's usually best not to have defaults, if possible. This way if the user does not specify a value here, it means "use the default value the API uses on creation, and do not change the value on updating". This is necessary anyway when adding new fields here to this option (to avoid breaking changes), and it can be confusing if some options do have defaults while others do not.

Reply:
I kept them on **false** by default, to avoid accidentally turning on some required action, that might mess up the authentication in Keycloak. But, there's no need for it, since by default, Keycloak leaves the options on **false**, when the parameters are not set by the user. Therefore, I'm removing the defaults.